### ISSUE TITLE
[Bug]: Legacy CUDA RMSNorm quantization clips E5M2 output to the E4M3 range

### ISSUE THREAD

**Issue by morluto** [flashinfer-ai/flashinfer#4053]
## Observed behavior

With the legacy CUDA norm backend, `rmsnorm_quant` and `fused_add_rmsnorm_quant` clamp E5M2 output to ±448, the E4M3 finite range.

E5M2 supports finite values through ±57344, so representable results between 448 and 57344 are silently reduced to 448.

This affects `FLASHINFER_USE_CUDA_NORM=1` and fallback use when the CuTe DSL implementation is unavailable.

## Reproduction

```python
import torch
import flashinfer

x = torch.ones(1, 128, dtype=torch.float16, device="cuda")
weight = torch.full((128,), 1024.0, dtype=torch.float16, device="cuda")
out = torch.empty_like(x, dtype=torch.float8_e5m2)

flashinfer.rmsnorm_quant(out, x, weight, 1.0, enable_pdl=False)
print(out[0, 0].float())
```

For these inputs, RMS normalization produces approximately 1024:

| Output type | Finite maximum | Input to clamp | Previous output | Expected output |
|---|---:|---:|---:|---:|
| E4M3 | 448 | 1024 | 448 | 448 |
| E5M2 | 57344 | 1024 | 448 | 1024 |

```text
previous = clamp(1024, -448, 448) = 448
expected = clamp(1024, -57344, 57344) = 1024
```

## Root cause

Both legacy CUDA kernels use a hard-coded ±448 clamp even though the output type is available as a template parameter.

## Scope

Affected:

- Legacy CUDA `rmsnorm_quant`
- Legacy CUDA `fused_add_rmsnorm_quant`
- E5M2 results above 448

Not affected:

- E4M3 output
- INT8
- The default CuTe implementation

## Success criteria

- Both kernels select the limit from the output FP8 type.
- E4M3 continues to saturate at 448.
- E5M2 preserves representable values above 448.
- Both APIs and formats receive regression coverage.

**Comment by bkryu** [comment]
Hi @morluto, as FlashInfer phases out the CUDA backend, if we expand support of the cute-dsl kernel, would that solve the issue for you?

We could try to make the cute-dsl `rmsnorm_quant` and `fused_add_rmsnorm_quant` kernels support both fp16/bf16 inputs and e4m3/e5m2 outputs. The cute-dsl route should have better performance than the CUDA variant

**Comment by bkryu** [comment]
Expected to be fixed by #4202

**Comment by bkryu** [comment]
#4202 merged. Closing the issue. @morluto , please feel free to reopen the issue if it persists.
